### PR TITLE
Add float64 arithmetic

### PR DESCRIPTION
**Pull request by andrewmostello**
Add float64 arithmetic functions. For clarity, these functions all have the "f" suffix. To avoid precision errors, all floats are converted to a decimal representation value using [shopspring/decimal](https://github.com/shopspring/decimal) before calculating.

### CODE REVIEW

File: go.mod
Comment (mattfarina) [inline go.mod]: I was just looking at this package in depth and I have some concerns as it appears to be mostly unmaintained. For example:

* There are only two commits this calendar year
* Numerous bug fixes, improvements, and the move to modules are all sitting as PRs without responses
* There are bugs listed in the issue queue. I wonder if our users will run into these

The company that maintained this package was bought and the service that was using this appears to have been shut down. It looks to have issues while being unmaintained which gives me reason to pause.

I like what it does to make working with floats more practical.

Comment (technosophos) [reply go.mod]: Do any of the bugs impact this usage?

**Comment by andrewmostello** [comment]
Thanks for looking into it.

There's an issue open on the package concerning its maintenance, and it looks like it has been handed off to new maintainers if that helps resolve the matter: https://github.com/shopspring/decimal/issues/135

**Comment by Moser-ss** [comment]
Hey, any plans to merge this PR?
I am using sprig inside of the Helm template, and I had a case where I need to multiply an int64 to with  float64. In this case, I want to increase 1GB with 0.5 (that means cut the value in half), and I can't because the math functions don't accept float64 either return a float64.
What is blocking this PR, and exists something I can help to push this?

**Comment by andrewmostello** [comment]
A few quick updates on this PR:
- I just rebased my fork again, so the PR is the addition of one commit
- shopspring/decimal is actively being maintained, and they've been tagging releases since April
- I've updated the dependency to the latest tag release

I'm still hoping to get this (or similar functionality) into sprig. Please let me know if there's anything else I can do here.

Thanks!

**Comment by mattfarina** [comment]
@andrewmostello Thanks for continuing to work on this.

**Review by technosophos — APPROVED** [review]
LGTM. Thanks for seeing this through!

**Comment by andrewmostello** [comment]
No problem! I just rebased again to resolve the conflict.

**Review by mattfarina — APPROVED** [review]

**Comment by Moser-ss** [comment]
Question: @mattfarina @technosophos When this feature will be released? Because I am anxious to create a PR in the Helm project to upgrade this lib!

**Pull request merged**
